src/main/java/gregtech/common/pipelike/PipeCollectorWalker.java

```diff
@@ -0,0 +1,68 @@
+package gregtech.common.pipelike;
+
+import gregtech.api.pipenet.PipeNetWalker;
+import gregtech.api.pipenet.tile.IPipeTile;
+
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class PipeCollectorWalker<T extends IPipeTile<?, ?>> extends PipeNetWalker<T> {
+
+    @NotNull
+    @SuppressWarnings("unchecked")
+    public static List<IPipeTile<?, ?>> collectPipeNet(World world, BlockPos sourcePipe, IPipeTile<?, ?> pipe,
+                                                       int limit) {
+        PipeCollectorWalker<? extends IPipeTile<?, ?>> walker = (PipeCollectorWalker<? extends IPipeTile<?, ?>>) new PipeCollectorWalker<>(
+                world, sourcePipe, 0, pipe.getClass());
+        walker.traversePipeNet(limit);
+        return walker.isFailed() ? Collections.EMPTY_LIST : Collections.unmodifiableList(walker.pipes);
```

Comment:
This should return `Collections.emptyList()` instead of `Collections.EMPTY_LIST`. This change also means the `@SuppressWarnings("unchecked")` can be removed from this method.